### PR TITLE
Added LAG DB objects

### PR DESCRIPTION
**Pull request by niksv**

### CODE REVIEW

File: src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+        @Mapping(target = "lagLogicalPort", ignore = true)
+        void copyWithoutLagLogicalPort(PhysicalPortData source, @MappingTarget PhysicalPortData target);
+
+        /**
+         * Performs deep copy of entity data.
+         */
+        default PhysicalPortData deepCopy(PhysicalPortData source) {
+            PhysicalPortData result = new PhysicalPortDataImpl();
+            copyWithoutLagLogicalPort(source, result);
```
Comment (niksv) [inline src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java:149]: Here was line 
```result.setLagLogicalPort(new LagLogicalPort(source.getLagLogicalPort()));```
but this line causes stack overflow.

I'm not sure should it be here or not.
@sergii-iakovenko Please help Dasha to understand how to copy DB objects

maxresdefault

Comment (sergii-iakovenko) [reply src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java:149]: Please, check how it's done for FlowMirrorPath / FlowMirrorPoints. In short, you can't do this with the same copy constructor of LagLogicalPort. Instead, you need another one which doesn't clone composite PhysicalPort(s).

File: src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    @NoArgsConstructor
+    @AllArgsConstructor
+    static final class PhysicalPortDataImpl implements PhysicalPortData, Serializable {
+        private static final long serialVersionUID = 1L;
+        @NonNull SwitchId switchId;
+        int portNumber;
+
+        @ToString.Exclude
+        @EqualsAndHashCode.Exclude
+        @NonNull LagLogicalPort lagLogicalPort;
```
Comment (sergii-iakovenko) [inline src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java:129]: But will the setter method update a back reference in LagLogicalPort? I think, no.

File: src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/LagLogicalPortFrame.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Slf4j
+public abstract class LagLogicalPortFrame extends KildaBaseVertexFrame implements LagLogicalPortData {
+    public static final String FRAME_LABEL = "lag_logical_port";
+    public static final String OWNS_PHYSICAL_PORT_EDGE = "owns";
```
[the remark this answers is not in the document]
Comment (niksv) [reply src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/LagLogicalPortFrame.java:39]: renamed

File: src-java/kilda-model/src/main/java/org/openkilda/model/LagLogicalPort.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            return Collections.unmodifiableList(physicalPorts);
+        }
+
+        @Override
+        public void setPhysicalPorts(List<PhysicalPort> physicalPorts) {
+            this.physicalPorts = new ArrayList<>(physicalPorts);
+            for (PhysicalPort port : this.physicalPorts) {
+                if (port.getData() instanceof PhysicalPortDataImpl) {
+                    ((PhysicalPortDataImpl) port.getData()).lagLogicalPort = this.lagLogicalPort;
+                }
```
[the remark this answers is not in the document]
Comment (niksv) [reply src-java/kilda-model/src/main/java/org/openkilda/model/LagLogicalPort.java:168]: fixed

File: src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java (hunk 2 above)
Comment (niksv) [reply src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java:129]: fixed

File: src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java (hunk 1 above)
Comment (niksv) [reply src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java:149]: fixed

File: src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/LagLogicalPortFrame.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (niksv) [reply src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/LagLogicalPortFrame.java:39]: renamed

File: src-java/kilda-model/src/main/java/org/openkilda/model/LagLogicalPort.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        data = new LagLogicalPortDataImpl();
+    }
+
+    /**
+     * Cloning constructor which performs deep copy of the entity.
+     *
+     * @param entityToClone the LAG logical port entity to copy data from.
+     */
+    public LagLogicalPort(@NonNull LagLogicalPort entityToClone) {
+        this();
```
Comment (nikitamarchenko) [inline src-java/kilda-model/src/main/java/org/openkilda/model/LagLogicalPort.java:71]: Why do we need this call?

**Review by nikitamarchenko — APPROVED** [review]

**Review by sergii-iakovenko — APPROVED** [review]

**Pull request merged**